### PR TITLE
Migrate `dependencies.yml` to `dependencies.toml`

### PR DESCRIPTION
**Pull request by ikhoon**
Motivation:

Gradle's version catalogs provides concise and sophisticated version
manangement.

Modifications:

- Switch `dependencies.yml` into `dependencies.toml`
- Migrate all `build.gradle`s to use new version catalogs API
- Fix wrong dependency in `spring-boot{1}-autoconfigure`
  - Drop `spring-boot-starter` dependency and correctly depend on `spring-boot-autoconfigure`
- Make `managedVersions` task generate a backward compatible `managed_versions.yml` file.

Result:

Better dependency manangement.

### CODE REVIEW

File: dependencies.toml [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+spring-boot2 = "2.7.3"
+
+[boms]
+armeria = { module = "com.linecorp.armeria:armeria-bom", version = "1.19.0" }
+jackson = { module = "com.fasterxml.jackson:jackson-bom", version = "2.13.4" }
+junit5 = { module = "org.junit:junit-bom", version = "5.9.0" }
+
+[libraries.armeria]
+module = "com.linecorp.armeria:armeria"
+javadocs = "https://www.javadoc.io/doc/com.linecorp.armeria/armeria-javadoc/1.18.0/"
```
Comment (minwoox) [inline dependencies.toml:24]:
```suggestion
javadocs = "https://www.javadoc.io/doc/com.linecorp.armeria/armeria-javadoc/1.19.0/"
```

**Review by minwoox — APPROVED** [review]
Thanks! 👍

**Comment by jrhee17** [comment]
Seems like there is a genuine failure.  Can you check? 😅 

```
FAILURE: Build failed with an exception.

* Where:
Script '/home/runner/work/centraldogma/centraldogma/gradle/scripts/lib/common-dependencies.gradle' line: 170

* What went wrong:
A problem occurred evaluating script.
> No signature of method: java.util.Optional.isEmpty() is applicable for argument types: () values: []
  Possible solutions: empty(), identity(groovy.lang.Closure), inspect(), dump()
```

**Review by jrhee17 — APPROVED** [review]
Changes look good 👍  Can you check the build failure I mentioned?

**Comment by ikhoon** [comment]
I missed linking to the patch of the bug. 
It will be fixed by https://github.com/line/gradle-scripts/pull/128

**Comment by minwoox** [comment]
👍 👍 👍

**Pull request merged**